vyper/venom/basicblock.py

```diff
@@ -237,6 +241,34 @@ def __repr__(self):
         return json.dumps(self.value)  # escape it
 
 
+@dataclass(frozen=True)
+class MemoryLocation:
+    """Represents a memory location that can be analyzed for aliasing"""
+
+    offset: int = 0
+    size: int = 0
+    is_volatile: bool = False
+
+    def completely_overlaps(self, other: "MemoryLocation") -> bool:
+        if self == FULL_MEMORY_ACCESS:
+            return True
+        if other == FULL_MEMORY_ACCESS:
+            return self == FULL_MEMORY_ACCESS
+        if self == EMPTY_MEMORY_ACCESS or other == EMPTY_MEMORY_ACCESS:
```

Comment:
not sure this is right. if `other == EMPTY_MEMORY_ACCESS` we should return true, right?

Reply:
Only if we are FULL_MEMORY_ACCESS, which is handled in the beginning.